corehq/apps/app_manager/util.py

```diff
@@ -688,3 +689,21 @@ def get_form_source_download_url(xform):
         xform.build_id,
         app.get_form_filename(module=form.get_module(), form=form),
     ])
+
+
+@quickcache(['domain', 'profile_id'], timeout=24 * 60 * 60)
+def get_latest_enabled_build_for_profile(domain, profile_id):
+    from corehq.apps.app_manager.models import LatestEnabledBuildProfiles
+    latest_enabled_build = (LatestEnabledBuildProfiles.objects.
+                            filter(build_profile_id=profile_id)
+                            .order_by('-version')
+                            .first())
+    if latest_enabled_build:
+        return get_app(domain, latest_enabled_build.build_id)
+
+
+@quickcache(['build_id', 'version'], timeout=24 * 60 * 60)
+def get_enabled_build_profiles_for_version(build_id, version):
```

Comment:
Nit: I think it's redundant to check both `build_id` and `version`, isn't it?  Shouldn't a build_id imply a version?

Reply:
that's true. Just a safe keep in case of super rare case if two build profile ids be the same.